sklearn/utils/parallel.py

```diff
@@ -0,0 +1,104 @@
+"""Module that customize joblib tools for scikit-learn usage."""
+
+import functools
+import warnings
+from functools import update_wrapper
+
+import joblib
+
+from .._config import config_context, get_config
+
+
+def _with_config(delayed_func, config):
+    """Helper function that intends to attach a config to a delayed function."""
+    if hasattr(delayed_func, "with_config"):
+        return delayed_func.with_config(config)
+    else:
+        warnings.warn(
+            "You are using `sklearn.utils.parallel.Parallel` that intend to attach a "
+            "configuration to a delayed function. However, the function used for "
+            "delaying the function does not expose `with_config`. Use "
+            "`sklearn.utils.parallel.delayed` to correctly propagate the scikit-learn "
+            "configuration to the joblib workers.",
+            UserWarning,
+        )
+        return delayed_func
+
+
+class Parallel(joblib.Parallel):
```

Comment:
Since this is now public, let's add `Parallel` and `delayed` into https://github.com/scikit-learn/scikit-learn/blob/main/doc/modules/classes.rst ?